timely/src/dataflow/operators/exchange.rs

```diff
@@ -23,13 +23,34 @@ pub trait Exchange<T, D: ExchangeData> {
     /// });
     /// ```
     fn exchange(&self, route: impl Fn(&D)->u64+'static) -> Self;
+
+    /// Exchange records between workers.
+    ///
+    /// The closure supplied should map a reference to a record to a `u64`,
+    /// whose value determines to which worker the record will be routed.
+    ///
+    /// # Examples
+    /// ```
+    /// use timely::dataflow::operators::{ToStream, Exchange, Inspect};
+    /// use timely::dataflow::channels::pact::LazyExchange;
+    ///
+    /// timely::example(|scope| {
+    ///     (0..10).to_stream(scope)
+    ///            .exchange_pact(LazyExchange::new(|x| *x))
+    ///            .inspect(|x| println!("seen: {:?}", x));
+    /// });
+    /// ```
+    fn exchange_pact<P: ParallelizationContract<T, D>>(&self, pact: P) -> Self;
```

Comment:
This seems a bit weird. Not all of the pacts are `Exchange*` pacts. Maybe `apply_pact` as the operator name?

Reply:
Changed `exchange_pact` to `apply_pact`.